src/lib/components/PrismicImage.svelte

```diff
@@ -1,34 +1,63 @@
 <script lang="ts">
-	import * as prismicT from "@prismicio/types";
+	import * as prismicH from "@prismicio/helpers"
+	import type * as prismicT from "@prismicio/types";
+	
+	/**
+	 * A Prismic Image field.
+	 */
 	export let field: prismicT.ImageField;
-	let className: string;
 
-	// creates a `class` property, even
-	// though it is a reserved word
-	export { className as class };
+	/**
+	 * An object of Imgix URL API parameters to transform the image.
+	 *
+	 * See: https://docs.imgix.com/apis/rendering
+	 */
+	export let imgixParams: Parameters<typeof prismicH.asImageSrc>[1] | null | undefined = null;
 
-	const usePrismicImage = (
-		field: prismicT.ImageField,
-	): {
-		url: string;
-		width?: number;
-		height?: number;
-		alt?: string;
-		copyright: string | undefined;
-	} => {
-		field.url;
-		return {
-			url: field.url || "",
-			alt: field.alt || "",
-			width: field.dimensions?.width || undefined,
-			height: field.dimensions?.height || undefined,
-			copyright: field.copyright || undefined,
-		};
-	};
+	/**
+	 * Widths used to build a `srcset` value for the Image field.
+	 *
+	 * If a `widths` prop is not given or `"defaults"` is passed, the
+	 * following widths will be used: 640, 750, 828, 1080, 1200, 1920, 2048,
+	 * 3840.
+	 *
+	 * If the Image field contains responsive views, each responsive view
+	 * can be used as a width in the resulting `srcset` by passing
+	 * `"thumbnails"` as the `widths` prop.
+	 */
+	export let widths: NonNullable<Parameters<typeof prismicH.asImageWidthSrcSet>[1]>["widths"] | null | undefined = null;
+				
+	
+	/**
+	 * HTML class attribute. Use a `:global()` selector to target the class.
+	 */
+	let className: string | null | undefined = null;
+	export { className as class };
 
-	const image = usePrismicImage(field);
+	const params = {}
+	if(widths) Object.assign(params, { widths })
+	if(imgixParams) Object.assign(params, imgixParams)
 </script>
 
-{#if field.url}
-	<img src={image.url} alt={image.alt} height={image.height} />
-{/if}
+<!-- 
+  @component
+  Component to render a Prismic Image field as an `img` element with `width`, `height`, `alt`, `src`, and `srcset` attributes.
+  
+  @example Rendering an Image field:
+	```svelte
+		<PrismicImage 
+			field={document.data.example_image} 
+			imgixParams={{ sat: -100 }}
+		/>
+  ```
+-->
+
+{#if prismicH.isFilled.image(field)}
+	{@const { alt, dimensions } = field}
+	{@const { width, height } = dimensions }
+	{@const { src, srcset } = prismicH.asImageWidthSrcSet(field, params)}
+
+	<img {src} {srcset} {alt} {width} {height} class={className} />
```

Comment:
`width` and `height` should be overridable via props.

Reply:
✅

Reply:
I added a little logic so that the component will use either both `field.dimensions.width` and `field.dimensions.height` or neither. If a user provides `width` via a prop, they should get default HTML behavior, not `field.dimensions.height`.

Reply:
Good idea. Maybe we could combine that idea with what we do in `@prismicio/next`'s `<PrismicNextImage>` component. Since we know the image's aspect ratio, we can compute the images' width or height if we are only given one of them.

```ts
const ar = field.dimensions.width / field.dimensions.height;

const castedWidth = castInt(width);
const castedHeight = castInt(height);

let resolvedWidth = castedWidth ?? field.dimensions.width;
let resolvedHeight = castedHeight ?? field.dimensions.height;

if (castedWidth != null && castedHeight == null) {
	resolvedHeight = castedWidth / ar;
} else if (castedWidth == null && castedHeight != null) {
	resolvedWidth = castedHeight * ar;
}
```
Source: https://github.com/prismicio/prismic-next/blob/cc825bd8ca36b4215880d61c7774ca9c10552f4b/src/PrismicNextImage.tsx#L109-L121

`resolvedWidth` and `resolvedHeight` should be correctly sized after the `if...else` statement.

`castInt()` is a helper to coerce a value into a number or `undefined`:

```ts
const castInt = (input: string | number | undefined): number | undefined => {
	if (typeof input === "number" || typeof input === "undefined") {
		return input;
	} else {
		const parsed = Number.parseInt(input);

		if (Number.isNaN(parsed)) {
			return undefined;
		} else {
			return parsed;
		}
	}
};
```
Source: https://github.com/prismicio/prismic-next/blob/cc825bd8ca36b4215880d61c7774ca9c10552f4b/src/PrismicNextImage.tsx#L11-L23

Reply:
Cool, thanks! I've done an implementation 👍